returnn/__main__.py

```diff
@@ -228,7 +228,7 @@ def load_data(config, cache_byte_size, files_config_key, **kwargs):
     Dataset.kwargs_update_from_config(config, kwargs)
     data = init_dataset(kwargs)
   elif config.is_typed(files_config_key) and callable(config.typed_value(files_config_key)):
-    data = init_dataset(config.typed_value(files_config_key))
+    data = init_dataset(config.typed_value(files_config_key), default_kwargs=kwargs)
```

Comment:
Doesn't `extra_kwargs` make more sense? What do you have in `kwargs` here?

Reply:
`kwargs` in this case are the globally defined dataset arguments from `Dataset.get_default_kwargs_eval(config=config)`.

In the case that the output of the callable is a Dataset, both `default_kwargs` and `extra_kwargs` are currently not used.

If the output of the callable is a dict, `extra_kwargs` would overwrite the values in the dict with these global default values. So `default_kwargs` should be correct here.

Reply:
I just notice that there are more problems here. In case that the callable returns a dict, I think we want that it applies the logic from above, esp the logic involving `kwargs`.

Reply:
Maybe it's easier if I just fix this in a follow-up PR.

Reply:
Although, maybe it's not so important... What you miss here is the `Dataset.kwargs_update_from_config`. Not sure...